js/client-mainmenu.js

```diff
@@ -1318,7 +1318,7 @@
 				);
 			}
 			var html = '';
-			if (!bufs.length) {
+			if (!bufs.every(function (buf) { return !!buf; })) {
```

Comment:
This logic is wrong, which is why you added the length check to try to fix it.

Reply:
I still need the length check in case `bufs` is empty

Reply:
actually nvm I see what you mean now